cabot/cabotapp/alert.py

```diff
@@ -23,17 +25,43 @@ def send_alert(self, service, users, duty_officers):
         """
         return True
 
+    def send_test_alert(self, user):
+        """
+        Send a test alert when the user requests it (to make sure config is valid).
+
+        The default implementation creates a fake Service and StatusCheck and calls the normal send_alert().
+        Note that the service/check may not be fully configured (e.g. invalid primary key, no HipChat room ID, ...).
+        :param user: django user
+        :return: nothing, raise exceptions on error
+        """
+        service_mock = create_failing_service_mock()
+        self.send_alert(service_mock, [], [user])
+
 
 class AlertPluginUserData(PolymorphicModel):
     title = models.CharField(max_length=30, editable=False)
     user = models.ForeignKey('UserProfile', editable=False)
 
+    # This is used to add the "Send Test Alert" button to the edit page.
+    # We need this information to be able to map AlertPluginUserData subclasses to their AlertPlugins.
+    # It's a list because some plugins (like Twilio) have multiple alert types for one user data type.
+    alert_classes = []
+
     class Meta:
         unique_together = ('title', 'user',)
 
     def __unicode__(self):
         return u'%s' % (self.title)
 
+    def is_configured(self):
```

Comment:
Do any of our plugins need this function to be overridden?

Reply:
Yeah. Like I said, this is based on the "test alerts" branch, since that branch changes the profile code quite a bit. This should only get merged after the "test alerts" branch does.